README.md

```diff
@@ -0,0 +1,330 @@
+  .resolve("G:A:V").withTransitivity().asFile();
+
+Maven.configureResolver().withRemoteRepo(MavenRemoteRepositories.createRemoteRepository("my-repository-id", "url://to/my/repository", "layout"))
+  .resolve("G:A:V").withTransitivity().asFile();
+```
+
+### Resolution of Artifacts from POM Files
+
+In Maven projects, you likely specify dependencies in your pom.xml file. ShrinkWrap Resolvers allows you to avoid redundancy by automatically loading metadata from there.
+
+- To load dependencies from the POM:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml").resolve("G:A").withTransitivity().asFile();
+  ```
+- Or, for dependencies with non-JAR packaging or classifier:
+  ```java
+  Maven.resolver().loadPomFromClassLoaderResource("/path/to/pom.xml").resolve("G:A:P:?").withTransitivity().asFile();
+  ```
+- To import and resolve dependencies based on specific scopes:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml").importDependencies(ScopeType.TEST, ScopeType.PROVIDED).resolve().withTransitivity().asFile();
+  ```
+- Or use convenience methods:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml").importRuntimeDependencies().resolve().withTransitivity().asFile();
+  ```
+
+- To activate profiles:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml", "activate-profile-1", "!disable-profile-2").importRuntimeAndTestDependencies().resolve().withTransitivity().asFile();
+  ```
+
+### Version Range Resolution
+The ShrinkWrap Resolver API allows for resolution of available versions info from a requested range. The [Maven documentation](http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html) specifies the version range syntax.
+```java
+final MavenVersionRangeResult versionRangeResult = Maven.resolver().resolveVersionRange("G:A:[1.0.0]");
+```
+
+### System properties
+
+ShrinkWrap Resolvers allows you to override any programmatic configuration via System properties.
+
+- `org.apache.maven.user-settings`
+- `org.apache.maven.global-settings`
+- `settings.security`
+- `org.apache.maven.offline`
+- `org.apache.maven.flattened-pom-path`
+- `maven.repo.local`
+- `maven.legacyLocalRepo`
+
+
+## Embedded Maven
+
+Embedded Maven allows direct invocation of Maven builds from Java code, offering functionalities such as downloading and using desired Maven binaries, an uncluttered API for simple or complex builds, additional methods for build control (e.g., ignoring failures), and easy access to build outputs and artifacts. It can be invoked using ShrinkWrap Resolver API or custom Invoker and InvocationRequest instances.
+
+### Downloading Maven Binaries
+
+Embedded Maven facilitates downloading and using specific Maven versions from Apache web pages, caching downloaded binaries, and setting up Maven home for the build.
+
+```java
+EmbeddedMaven.forProject("path/to/pom.xml").useMaven3Version(String version)
```

Comment:
I would either used comments or explained this code block a little bit. For example, I'm missing the information about using cached Maven